### ISSUE TITLE
Every time we need to manually edit the YAML and remove the credential and key part so we can add backup-location and work as excpected

### ISSUE THREAD

**Issue by marc-fekry** [vmware-tanzu/velero#4576]
customer trying to add backup-location as follow:
sudo velero --kubeconfig=.kube/config backup-location create slave2 \
   --provider aws \
   --bucket jedox-prod-bucket \
   --prefix XXXXXXX/ \
   --access-mode ReadOnly \
   --default \
   --config bucket=XXXXX \
   --config region=XXXX \
   --config profile=XXXXX \
   --config publicUrl=http://1XXXX:9000 ; \
   --config s3ForcePathStyle=true \
   --config s3Url="http://XXXX:9000"
   
    Unfortunately in the spec part in the YAML file we see credential and key that are not intend to be there.
```
spec:
  accessMode: ReadOnly
  credential:
    key: ""
```

We need a way to do this without editing the YAML (Using the command line)

when we use command line --help we see in the optional part:
--credential mapStringString  The credential to be used by this location as a key-value pair, where the key is the Kubernetes Secret name, and the value is the data key name within the Secret. Optional, one value only.
From the doc: https://velero.io/docs/v1.6/locations/
  --credential=secret-name=key-within-secret(Which is not clear as well)

sudo velero --kubeconfig=.kube/config version

Client:
        Version: v1.7.0
        Git commit: "9e52260568430ecb77ac38a677ce74267a8c2176"
Server:
        Version: v1.6.3
WARNING: the client version does not match the server version. Please update server

**Comment by ywk253100** [comment]
For the `--credential` option of `velero backup-location create`, you need to create a `Kubernetes Secret` which contains the credential content first, and after that specifying the secret name and data key in the option.

For example, I have a secret named `cloud-credentials` as following
```
kubectl get secret cloud-credentials -o yaml -n velero
apiVersion: v1
data:
  cloud: Qjliklaaxxxx
kind: Secret
metadata:
  name: cloud-credentials
  namespace: velero
type: Opaque
```
I should specify the `--credential` option as `--credential=cloud-credentials=cloud`

**Comment by ywk253100** [comment]
@marc-fekry Your backup-location doesn't need credentials right? If this is the case, what error did you get if the yaml file contains the empty credential? Could you provide the logs to help me locate the root cause?
Maybe we should ignore the credential if it is empty

**Comment by zubron** [comment]
Hi @marc-fekry - Sorry you're having this problem. This is a known issue (https://github.com/vmware-tanzu/velero/issues/4196) and the fix will be available in 1.8 which will be released in February. If you would like to try the RC, it is available here: https://github.com/vmware-tanzu/velero/releases/tag/v1.8.0-rc.1.

I am going to close this as it has already been fixed.
